src/renderer/vt/Xterm256Engine.cpp

```diff
@@ -85,28 +85,13 @@ Xterm256Engine::Xterm256Engine(_In_ wil::unique_hfile hPipe,
         _lastTextAttributes.SetFaint(true);
     }
 
-    // Turning off the underline styles must be handled at the same time,
-    // since there is only one sequence that resets both of them.
-    const auto singleTurnedOff = !textAttributes.IsUnderlined() && _lastTextAttributes.IsUnderlined();
-    const auto doubleTurnedOff = !textAttributes.IsDoublyUnderlined() && _lastTextAttributes.IsDoublyUnderlined();
-    if (singleTurnedOff || doubleTurnedOff)
-    {
-        RETURN_IF_FAILED(_SetUnderlined(false));
-        _lastTextAttributes.SetUnderlined(false);
-        _lastTextAttributes.SetDoublyUnderlined(false);
-    }
-
-    // Once we've handled the cases where they need to be turned off,
-    // we can then check if either should be turned back on again.
-    if (textAttributes.IsUnderlined() && !_lastTextAttributes.IsUnderlined())
-    {
-        RETURN_IF_FAILED(_SetUnderlined(true));
-        _lastTextAttributes.SetUnderlined(true);
-    }
-    if (textAttributes.IsDoublyUnderlined() && !_lastTextAttributes.IsDoublyUnderlined())
+    // We check the singly, doubly underlined and extended styling together,
+    // since only one of them can be active at a time.
+    const auto ulStyle = textAttributes.GetUnderlineStyle();

```

Comment:
This is an interesting case actually. I'm betting it's *okay* because of the prior discussion about merging single and double and how there's only one terminal that keeps them separate (xterm).

We no longer turn off Double when we turn on Single and vice-versa!

Again, probably fine. Just an interesting caveat. :smile:

Reply:
You're right! I probably missed it. 😅👍

I think, to keep pty client in sync with our "one style at a time" impl, we should first send `"\e[24m"` before sending the new style. This would ensure the client never holds any stale singly or doubly style

Reply:
I made some changes. vtrenderer would now send `"\e[24m"` before changing the underline style from doubly underlined.